async-gateway/service.go

```diff
@@ -66,7 +66,7 @@ func (s *service) CreateWorkload(id string, payload io.Reader, contentType strin
 		return "", err
 	}
 
-	statusPath := fmt.Sprintf("%s/%s/status", prefix, id)
+	statusPath := fmt.Sprintf("%s/%s/%s", prefix, id, StatusInQueue)
 	log.Debug(fmt.Sprintf("setting status to %s", StatusInQueue))
 	if err := s.storage.Upload(statusPath, strings.NewReader(string(StatusInQueue)), "text/plain"); err != nil {
```

Comment:
Nit: if we aren't checking the contents of the file, it might be more clear to the reader if we create an empty file rather than having the file contain the status